packages/tools/client-debugger/client-debugger/src/messaging/Utilities.ts

```diff
@@ -16,20 +17,24 @@ import { ISourcedDebuggerMessage } from "./Messages";
  *
  * @internal
  */
-export function postMessagesToWindow<TMessage extends ISourcedDebuggerMessage>(
+export function postMessagesToWindow<TMessage extends IDebuggerMessage>(
 	loggingOptions?: MessageLoggingOptions,
 	...messages: TMessage[]
 ): void {
+	const messagesWithSource: ISourcedDebuggerMessage[] = messages.map((m) => ({
+		...m,
+		source: debuggerMessageSource,
+	}));
+
 	// TODO: remove loggingOptions once things settle.
 	// If we need special logic for globalThis.postMessage maybe keep this function, but otherwise maybe remove it too.
 	if (loggingOptions !== undefined) {
 		const loggingPreamble =
 			loggingOptions?.context === undefined ? "" : `${loggingOptions.context}: `;
-		console.debug(`${loggingPreamble}Posting message to the window:`, messages);
+		console.debug(`${loggingPreamble}Posting messages to the window:`, messagesWithSource);
```

Comment:
I think if you want the contents object to actually be visible you might have to take it out of the backtick string, unless console.debug prints differently than console.log. Otherwise this might print "messages to the window: [object]".

Reply:
It's being passed as second parameter, so it gets serialized in both cases (console.log and console.debug()):
<img width="201" alt="image" src="https://user-images.githubusercontent.com/716334/228321882-f914228c-4f98-444b-b78d-f23332f98b94.png">